src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy

```diff
@@ -285,35 +278,31 @@ class MirrorEndpointsSpec extends HealthCheckSpecification {
     def "Can create mirror point on a VXLAN flow [#swPair.src.hwSwString, #mirrorDirection]#trafficDisclaimer"() {
         given: "A VXLAN flow"
         assumeTrue(swPair as boolean, "Unable to find required vxlan-enabled switches with traffgens")
-        def flow = flowHelperV2.randomFlow(swPair).tap { encapsulationType = FlowEncapsulationType.VXLAN }
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getBuilder(swPair)
+                .withEncapsulationType(FlowEncapsulationType.VXLAN)
+                .build().create()
 
         when: "Create a mirror point"
-        def mirrorTg = swPair.src.traffGens[1]
+        def mirrorTg = swPair.src.traffGens[1] // try to take TG at the same switch but not used in the flow
```

Comment:
Due to this comment it should be something like that 
`swPair.src.traffGens.find {  it.switchPort != flow.source.portNumber }`

Reply:
Fixed